### ISSUE TITLE
Assume role seems to fail with access denied

### ISSUE THREAD

**Issue by imrehg** [99designs/aws-vault#579]
On the latest aws-vault I can find on Mac (`v5.4.3`), I cannot assume a role, where the same setup worked on the last v4 (`v4.7.1`) release just fine:

```shell
$ aws-vault exec XYZ --debug
2020/05/05 09:21:56 aws-vault v5.4.3
2020/05/05 09:21:56 [keyring] Considering backends: [keychain pass file]
2020/05/05 09:21:56 Loading config file /Users/user/.aws/config
2020/05/05 09:21:56 Parsing config file /Users/user/.aws/config
2020/05/05 09:21:56 [keyring] Querying keychain for service="aws-vault", keychain="aws-vault.keychain"
2020/05/05 09:21:56 [keyring] Found 7 results
2020/05/05 09:21:56 [keyring] Querying keychain for service="aws-vault", keychain="aws-vault.keychain"
2020/05/05 09:21:56 [keyring] Found 7 results
2020/05/05 09:21:56 profile ABC: using stored credentials
2020/05/05 09:21:56 profile ABC: skipping GetSessionToken because profile 'XYZ' has no MFA serial defined
2020/05/05 09:21:56 profile XYZ: using AssumeRole
2020/05/05 09:21:56 Looking up keyring for 'ABC'
2020/05/05 09:21:56 [keyring] Querying keychain for service="aws-vault", account="ABC", keychain="aws-vault.keychain"
2020/05/05 09:22:00 [keyring] Found item "aws-vault (ABC)"
aws-vault: error: exec: Failed to get credentials for XYZ: AccessDenied: User: arn:aws:iam::<redacted>:user/<redacted> is not authorized to perform: sts:AssumeRole on resource: arn:aws:iam::<redacted>:role/<redacted>
	status code: 403, request id: 5f0d09bc-47d9-45f3-9802-647c1bafa0b5
```

Downgrading to `v4.7.1` everything works:

```shell
$ aws-vault exec XYZ --debug
2020/05/05 09:23:09 [keyring] Considering backends: [keychain pass file]
2020/05/05 09:23:09 Loading config file /Users/user/.aws/config
2020/05/05 09:23:09 Parsing config file /Users/user/.aws/config
2020/05/05 09:23:09 Looking for sessions for XYZ
2020/05/05 09:23:09 Looking up all keys in keyring
2020/05/05 09:23:09 [keyring] Querying keychain for service="aws-vault", keychain="aws-vault.keychain"
2020/05/05 09:23:09 [keyring] Found 7 results
2020/05/05 09:23:09 Session not found in keyring for XYZ
2020/05/05 09:23:09 Looking for sessions for ABC
2020/05/05 09:23:09 Looking up all keys in keyring
2020/05/05 09:23:09 [keyring] Querying keychain for service="aws-vault", keychain="aws-vault.keychain"
2020/05/05 09:23:09 [keyring] Found 7 results
2020/05/05 09:23:09 Session not found in keyring for ABC
2020/05/05 09:23:09 Looking up keyring for ABC
2020/05/05 09:23:09 [keyring] Querying keychain for service="aws-vault", account="ABC", keychain="aws-vault.keychain"
2020/05/05 09:23:11 [keyring] Found item "aws-vault (ABC)"
Enter token for arn:aws:iam::<redacted>:mfa/<redacted>: 123456
2020/05/05 09:23:14 Using region "eu-west-2" from profile
2020/05/05 09:23:14 Getting new session token for profile ABC
2020/05/05 09:23:15 Writing session for ABC to keyring: "session,bmhz,<redacted>,<redacted>"
2020/05/05 09:23:15 [keyring] Checking keychain status
2020/05/05 09:23:15 [keyring] Keychain status returned nil, keychain exists
2020/05/05 09:23:15 [keyring] Keychain item trusts keyring
2020/05/05 09:23:15 [keyring] Adding service="aws-vault", label="aws-vault session for ABC", account="session,bmhz,<redacted>,<redacted>", trusted=true to osx keychain "aws-vault.keychain"
2020/05/05 09:23:15 Using session ****************DG4J, expires in 3h59m59.588655s
2020/05/05 09:23:15 Using session ****************DG4J, expires in 3h59m59.58862s
2020/05/05 09:23:15 Using region "eu-west-2" from profile
2020/05/05 09:23:15 Assuming role arn:aws:iam::<redacted>:role/<redacted> from session token
2020/05/05 09:23:15 Using role ****************GIMR (from session token), expires in 14m59.450633s
2020/05/05 09:23:15 Setting subprocess env: AWS_DEFAULT_REGION=eu-west-2, AWS_REGION=eu-west-2
2020/05/05 09:23:15 Setting subprocess env: AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY
2020/05/05 09:23:15 Setting subprocess env: AWS_SESSION_TOKEN, AWS_SECURITY_TOKEN
```

The relevant part of `~/.aws/config`:
```
[profile ABC]
region=eu-west-2
mfa_serial=arn:aws:iam::<redacted>:mfa/<redacted>

[profile XYZ]
region=eu-west-2
source_profile=ABC
role_arn=arn:aws:iam::<redacted>:role/<redacted>
```

I also tried using `--mfa-token` manually, but that doesn't get used either.

Any ideas would be appreciated, including whether I'm using `aws-vault` wrong after the switch to v5.

- [x] I am using the latest release of AWS Vault
- [x] I have provided my `.aws/config` (redacted if necessary)
- [x] I have provided the debug output using `aws-vault --debug` (redacted if necessary)

**Comment by mtibben** [comment]
Hey @imrehg, see https://github.com/99designs/aws-vault/blob/master/USAGE.md#gotchas-with-mfa-config

TLDR; you'll need to define your `mfa_serial` in profile XYZ

**Comment by imrehg** [comment]
@mtibben thanks a lot, yeah, intuitiveness is relative. And I guess I just haven't looked at all the docs!
